sycl/test-e2e/FreeFunctionKernels/accessor_as_kernel_parameter.cpp

```diff
@@ -0,0 +1,187 @@
+    Queue.submit([&](sycl::handler &Handler) {
+      sycl::accessor<int, Dims> Accessor{Buffer, Handler};
+      Handler.set_args(Accessor, ExpectedResultValue);
+      Handler.single_task(UsedKernel);
+    });
+  }
+  return performResultCheck(NumOfElementsPerDim.size(), ResultData.data(),
+                            ErrorMessage, ExpectedResultValue);
+}
+template <auto Func, size_t Dims>
+int runNdRangeTest(sycl::queue &Queue, sycl::context &Context,
+                   sycl::nd_range<Dims> NdRange, std::string_view ErrorMessage,
+                   const int ExpectedResultValue) {
+  sycl::kernel UsedKernel = getKernel<Func>(Context);
+  std::vector<int> ResultData(NdRange.get_global_range().size(), 0);
+  {
+    sycl::buffer<int, Dims> Buffer(ResultData.data(),
+                                   NdRange.get_global_range());
+    Queue.submit([&](sycl::handler &Handler) {
+      sycl::accessor<int, Dims> Accessor{Buffer, Handler};
+      Handler.set_args(Accessor, ExpectedResultValue);
+      Handler.parallel_for(NdRange, UsedKernel);
+    });
+  }
+  return performResultCheck(NdRange.get_global_range().size(),
+                            ResultData.data(), ErrorMessage,
+                            ExpectedResultValue);
+}
+
+template <auto Func, size_t Dims>
+int runNdRangeTestMultipleParameters(sycl::queue &Queue, sycl::context &Context,
+                                     sycl::nd_range<Dims> NdRange,
+                                     std::string_view ErrorMessage,
+                                     sycl::range<3> Values) {
+  sycl::kernel UsedKernel = getKernel<Func>(Context);
+  std::vector<int> InputAData(NdRange.get_global_range().size(), Values[0]);
+  std::vector<int> InputBData(NdRange.get_global_range().size(), Values[1]);
+  std::vector<int> ResultData(NdRange.get_global_range().size(), 0);
+
+  {
+    sycl::buffer<int, Dims> InputABuffer(InputAData.data(),
+                                         NdRange.get_global_range());
+    sycl::buffer<int, Dims> InputBBuffer(InputBData.data(),
+                                         NdRange.get_global_range());
+    sycl::buffer<int, Dims> ResultBuffer(ResultData.data(),
+                                         NdRange.get_global_range());
+    Queue.submit([&](sycl::handler &Handler) {
+      sycl::accessor<int, Dims, sycl::access::mode::read,
+                     sycl::access::target::device>
+          InputAAcc{InputABuffer, Handler};
+      sycl::accessor<int, Dims, sycl::access::mode::read,
+                     sycl::access::target::device>
+          InputBAcc{InputBBuffer, Handler};
+      sycl::accessor<int, Dims, sycl::access::mode::write> ResultAcc{
+          ResultBuffer, Handler};
+      Handler.set_args(InputAAcc, InputBAcc, ResultAcc);
+      Handler.parallel_for(NdRange, UsedKernel);
+    });
+  }
+  return performResultCheck(NdRange.get_global_range().size(),
+                            ResultData.data(), ErrorMessage, Values[2]);
+}
+int main() {
+
```

Comment:
Nit; To me, inconsistent spacing like this makes the code hard to read. First glance makes my brain think `int main() {` is part of the function above. I would have a space between functions. There are a few other cases in the file, but some functions have spaces between them, so not all of them are affected.

```suggestion
}

int main() {
```